og.module

```diff
@@ -185,10 +181,19 @@ function og_entity_create_access(AccountInterface $account, array $context, $bun
     return $access_result;
   }
 
-  $node_access_strict = \Drupal::config('og.settings')->get('node_access_strict');
-  if ($entity_type_id == 'node' && !$node_access_strict && $account->hasPermission("create $bundle content")) {
-    // The user has the core permission and strict node access is not set.
-    return AccessResult::neutral();
+  $entity_access_strict = \Drupal::config('og.settings')->get('entity_access_strict');
+  if (!$entity_access_strict) {
+    if ($entity_type_id == 'node' && $account->hasPermission("create $bundle content")) {
+      // The user has the core permission and strict entity access is not set.
+      return AccessResult::neutral();
+    }
+
+    // Handle other entities.
+    if ($account->hasPermission("create $entity_type_id") ||
+        $account->hasPermission("create {$bundle} {$entity_type_id}")) {
+      // The user has the core permission and strict entity access is not set.
+      return AccessResult::neutral();
```

Comment:
This is problematic :( The permissions are just arbitrary strings so we cannot guarantee this to work.

This will work for a number of cases, but not all. I did a quick search in core, and it wouldn't work with the Book module for example (permission is `create new books`).

Until we have a structured way of handling permissions we'll have to allow modules to provide their own data. It seems this is being worked on in [Issue #2809177: Introduce entity permission providers and access handlers]()https://www.drupal.org/project/drupal/issues/2809177) but that is not in yet.

I think a good way to tackle this is to provide an event for this, and subscribe to it ourselves to return the list of permissions provided by the core modules. Developers can then implement a subscriber to provide any missing permissions.

Reply:
+1 on the event idea.

Reply:
Seems we already have a better handling of entity permissions in OG -- https://github.com/Gizra/og/issues/510#issuecomment-496440783